components/calendar/src/gregorian.rs

```diff
@@ -228,8 +228,109 @@ pub(crate) fn year_as_gregorian(year: i32) -> types::FormattableYear {
     } else {
         types::FormattableYear {
             era: types::Era(tinystr!(16, "bce")),
-            number: 1 - year,
+            number: 1_i32.saturating_sub(year),
             related_iso: None,
         }
     }
 }
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn day_of_year_info_max() {
+        #[derive(Debug)]
+        struct MaxCase {
+            year: i32,
+            month: u8,
+            day: u8,
+            next_year: i32,
+        }
+        let cases = [
+            MaxCase {
+                year: i32::MAX,
+                month: 7,
+                day: 11,
+                next_year: i32::MAX,
+            },
+            MaxCase {
+                year: i32::MAX,
+                month: 7,
+                day: 12,
+                next_year: i32::MAX,
+            },
+            MaxCase {
+                year: i32::MAX,
+                month: 8,
+                day: 10,
+                next_year: i32::MAX,
+            },
+            MaxCase {
+                year: i32::MAX - 1,
+                month: 7,
+                day: 11,
+                next_year: i32::MAX,
+            },
+        ];
+
+        for case in cases {
+            let date = Date::try_new_gregorian_date(case.year, case.month, case.day).unwrap();
+
+            assert_eq!(
+                Calendar::day_of_year_info(&Gregorian, &date.inner)
+                    .next_year
+                    .number,
+                case.next_year,
+                "{case:?}",
+            );
+        }
+    }
+    #[test]
+    fn day_of_year_info_min() {
+        #[derive(Debug)]
+        struct MinCase {
+            year: i32,
+            month: u8,
+            day: u8,
+            prev_year: i32,
+        }
+        let cases = [
+            MinCase {
+                year: i32::MIN,
+                month: 1,
+                day: 1,
+                prev_year: i32::MIN,
+            },
+            MinCase {
+                year: i32::MIN,
+                month: 12,
+                day: 31,
+                prev_year: i32::MIN,
+            },
+            MinCase {
+                year: i32::MIN,
+                month: 2,
+                day: 2,
+                prev_year: i32::MIN,
+            },
+            MinCase {
+                year: i32::MIN + 1,
+                month: 1,
+                day: 1,
+                prev_year: i32::MIN,
+            },
+        ];
+
+        for case in cases {
+            let date = Date::try_new_gregorian_date(case.year, case.month, case.day).unwrap();
+
+            assert_eq!(
+                Calendar::day_of_year_info(&Gregorian, &date.inner)
+                    .prev_year
+                    .number,
+                case.prev_year.saturating_neg(),
```

Comment:
Nit: If you always call `.saturating_neg()` on `case.prev_year`, then you should just represent `prev_year` in that form in the test case (as a positive year number).